### PR TITLE
Clarify app manager method names

### PR DESCRIPTION
**Pull request by come-nc**
* Resolves: # 

## Summary

Renames isInstalled -> isEnabledForAnyone, and getInstalledApps -> getEnabledApps.

There is a weird situation for apps which are enabled only for some groups of users:
- They are in the return of getInstalledApps/getEnabledApps
- They return true for isInstalled/isEnabledForAnyone (which is why I named it that way)
For isEnabled*, one can use isEnabledForUser($app, null), to check if the app is enabled for everyone.
But for getEnabled*, there is no equivalent, as getEnabledForUser parameter is mandatory.
Because of that, in NavigationManager, isEnabledForUser is called on each app returned by getInstalledApps.

We could consider that when there is no user in session we should always exclude apps enabled only for some groups, and I think for public pages it makes sense (does it?).
But the issue is for `occ` cli tool, there there is no user in session but we do want to consider app enabled for some groups only as enabled.

So basically for each getInstalledApps call, we should ask ourselves whether the list should include partially-enabled apps or not, and replace with the appropriate method call.
We may want to rename getEnabledApps with a more precise name, but getEnableAppsForAnyone is ugly, no?
We may want to add a way to get apps enabled for everyone, either as a new method or by making the parameter nullable in getEnabledAppsForUser.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [ ] Screenshots before/after for front-end changes
- [ ] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Comment by come-nc** [comment]
@nickvergessen Thoughts on the naming and logic?

File: lib/public/App/IAppManager.php [hunk 1]
```diff
@@ -178,9 +186,18 @@ public function getEnabledAppsForUser(IUser $user);
...
 	 * @since 8.1.0
+	 * @deprecated 31.0.0 Use either {@see self::getEnabledApps} or {@see self::getEnabledAppsForUser}
 	 */
 	public function getInstalledApps();
 
+	/**
+	 * List all apps enabled, either for everyone or for specific groups only
+	 *
+	 * @return list<string>
+	 * @since 31.0
```
Comment (nickvergessen) [inline lib/public/App/IAppManager.php:197]:
```suggestion
	 * @since 31.0.0
```

**Review by nickvergessen — APPROVED** [review]

File: core/Command/App/Enable.php [hunk 2]
```diff
@@ -77,20 +78,17 @@ private function enableApp(string $appId, array $groupIds, bool $forceEnable, Ou
 			return $group->getDisplayName();
 		}, $groupIds);
 
-		if ($this->appManager->isInstalled($appId) && $groupIds === []) {
+		if ($this->appManager->isEnabledForUser($appId) && $groupIds === []) {
```
[the remark this answers is not in the document]
Comment (come-nc) [reply core/Command/App/Enable.php:81]: This one is a bit technical, when there is no user in session, isEnabledForUser will only return true if the app is enabled for everyone (so, not only for some groups).

This is a behavior change, it means when doing :
```
occ app:enable myapp --groups mygroup
occ app:enable myapp
```

The second command will overwrite the first one and enable the app for everyone.
While in master it will fail with "myapp already enabled", and you have to do app:disable first.

I can revert that change if you think the change is a bad idea.

[the remark this answers is not in the document]
Comment (artonge) [reply core/Command/App/Enable.php:81]: No, I think it makes sense.
Can't we do `!isEnabledForAll($appId)`?

**Review by provokateurin — APPROVED** [review]
LGTM, just some of the commits should be squashed

File: core/Command/App/Enable.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (come-nc) [reply core/Command/App/Enable.php:81]: > No, I think it makes sense. Can't we do `!isEnabledForAnyone($appId)`?

No, it would not have the same result I think.

**Comment by come-nc** [comment]
rebased and squashed some commits

File: lib/private/App/AppManager.php [hunk 3]
```diff
@@ -129,32 +129,43 @@ private function getUrlGenerator(): IURLGenerator {
...
 
 	/**
-	 * List all installed apps
+	 * Deprecated alias
 	 *
 	 * @return string[]
 	 */
 	public function getInstalledApps() {
-		return array_keys($this->getInstalledAppsValues());
+		return $this->getEnabledApps();
```
Comment (blizzz) [inline lib/private/App/AppManager.php:159]: But this is not the same, right. An installed app might not be enabled. And an enabled app might not necessarily be installed (removed app folder for whatever reason, unfortunately i have one).

For example, when you re-create an instance by config, the enabled apps would not be installed anymore (since 30 actually), unless an upgrade is pending. 

OTOH i have not seen a another issue opened here on this matter, so maybe it really is too rare.

Comment (come-nc) [reply lib/private/App/AppManager.php:159]: This PR only change the name of the method, not what is returned?
It always returned a list of enabled applications.
Can you clarify what corner case was changed by this PR and how I can test it?
